ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp

```diff
@@ -592,9 +600,9 @@ MatmulProgramConfig get_matmul_program_config(
                 per_core_N_equals_subblock_w_constraint = true;
             }
 
-            uint32_t M = input_tensor_a.volume() / input_tensor_a.get_legacy_shape()[-1] / TILE_HEIGHT;
-            uint32_t K = input_tensor_a.get_legacy_shape()[-1] / TILE_WIDTH;
-            uint32_t N = input_tensor_b.get_legacy_shape()[-1] / TILE_WIDTH;
+            uint32_t M = input_tensor_a.volume() / input_tensor_a.get_legacy_shape()[-1] / in0_tile_shape[0];
+            uint32_t K = input_tensor_a.get_legacy_shape()[-1] / in0_tile_shape[1];
```

Comment:
In this file and in many other places we have this indexing by 0 and 1. I think some restructuring would make the code more readable, for example if you had in0_tile_shape[1] stored as width, or say if the shape could be accessed as a union to refer to the second element as width.